### PR TITLE
Upload multiple files at once

### PR DESCRIPTION
**Pull request by kvbik**
Functionality to upload multiple files at once.

The work was done in multiple steps:

* `requirements.txt` file for dependency management
* python code reorganised little bit to be better testable
* few unittests added: `python test_server.py`
* javascript logic changed to loop over multiple files

Got into many troubles with the for loop with `addEventListener`, but looks like I was able to resolve it.

### CODE REVIEW

**Comment by kvbik** [comment]
this obsoletes #2

File: static/js/app.js [hunk 1]
```diff
@@ -13,21 +13,31 @@ document.addEventListener('DOMContentLoaded', function() {
...
-		form.elements['filename'].value = fileList[0].name;
-		const reader = new FileReader();
-		reader.addEventListener('load', (event) => {
-			var arrayBuffer = event.target.result;
-			var fileData = new Uint8Array(arrayBuffer);
-			encryptFile(fileData).then(function(encrypted_file) {
-				form.elements['attachment'].value = encrypted_file;
+		for (var i=0; i < fileList.length; i++) {
+			current_file = fileList[i];
+			filename_field = form.elements['filename-' + i];
```
Comment (Aldekein) [inline static/js/app.js:23]: This throws `Uncaught TypeError: filename_field is undefined` in browser console if 6 files are selected.

**Review by Aldekein — APPROVED** [review]
The code looks good! We've discussed possible improvements in chat.

File: static/js/app.js (hunk 1 above)
Comment (kvbik) [reply static/js/app.js:23]: fixed in #4

**Pull request merged**